diff -Nru brotli-1.0.9.orig/c/common/platform.h brotli-1.0.9/c/common/platform.h
--- brotli-1.0.9.orig/c/common/platform.h	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/c/common/platform.h	2022-07-10 15:34:19.368776746 +0000
@@ -29,7 +29,7 @@
 #include <brotli/types.h>
 
 #if defined(OS_LINUX) || defined(OS_CYGWIN) || defined(__EMSCRIPTEN__)
-#include <endian.h>
+//#include <endian.h>
 #elif defined(OS_FREEBSD)
 #include <machine/endian.h>
 #elif defined(OS_MACOSX)
