diff -Nru brotli-1.0.9.orig/CMakeLists.txt brotli-1.0.9/CMakeLists.txt
--- brotli-1.0.9.orig/CMakeLists.txt	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/CMakeLists.txt	2022-07-10 17:05:29.565513161 +0000
@@ -137,7 +137,7 @@
 set(BROTLI_LIBRARIES ${BROTLI_LIBRARIES_CORE} ${LIBM_LIBRARY})
 mark_as_advanced(BROTLI_LIBRARIES)
 
-set(BROTLI_LIBRARIES_CORE_STATIC brotlienc-static brotlidec-static brotlicommon-static)
+set(BROTLI_LIBRARIES_CORE_STATIC brotlienc brotlidec brotlicommon)
 set(BROTLI_LIBRARIES_STATIC ${BROTLI_LIBRARIES_CORE_STATIC} ${LIBM_LIBRARY})
 mark_as_advanced(BROTLI_LIBRARIES_STATIC)
 
@@ -160,19 +160,21 @@
 transform_sources_list("scripts/sources.lst" "${CMAKE_CURRENT_BINARY_DIR}/sources.lst.cmake")
 include("${CMAKE_CURRENT_BINARY_DIR}/sources.lst.cmake")
 
-if(BROTLI_EMSCRIPTEN)
-  set(BROTLI_SHARED_LIBS "")
-else()
-  set(BROTLI_SHARED_LIBS brotlicommon brotlidec brotlienc)
-  add_library(brotlicommon SHARED ${BROTLI_COMMON_C})
-  add_library(brotlidec SHARED ${BROTLI_DEC_C})
-  add_library(brotlienc SHARED ${BROTLI_ENC_C})
-endif()
-
-set(BROTLI_STATIC_LIBS brotlicommon-static brotlidec-static brotlienc-static)
-add_library(brotlicommon-static STATIC ${BROTLI_COMMON_C})
-add_library(brotlidec-static STATIC ${BROTLI_DEC_C})
-add_library(brotlienc-static STATIC ${BROTLI_ENC_C})
+#if(1
+#  set(BROTLI_SHARED_LIBS "")
+#else()
+#  set(BROTLI_SHARED_LIBS brotlicommon brotlidec brotlienc)
+#  add_library(brotlicommon SHARED ${BROTLI_COMMON_C})
+#  add_library(brotlidec SHARED ${BROTLI_DEC_C})
+#  add_library(brotlienc SHARED ${BROTLI_ENC_C})
+#endif()
+
+set(BROTLI_SHARED_LIBS "")
+
+set(BROTLI_STATIC_LIBS brotlicommon brotlidec brotlienc)
+add_library(brotlicommon STATIC ${BROTLI_COMMON_C})
+add_library(brotlidec STATIC ${BROTLI_DEC_C})
+add_library(brotlienc STATIC ${BROTLI_ENC_C})
 
 # Older CMake versions does not understand INCLUDE_DIRECTORIES property.
 include_directories(${BROTLI_INCLUDE_DIRS})
@@ -195,13 +197,13 @@
   set_property(TARGET ${lib} APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${BROTLI_INCLUDE_DIRS}")
 endforeach()
 
-if(NOT BROTLI_EMSCRIPTEN)
+#if(NOT BROTLI_EMSCRIPTEN)
+#target_link_libraries(brotlidec brotlicommon)
+#target_link_libraries(brotlienc brotlicommon)
+#endif()
+
 target_link_libraries(brotlidec brotlicommon)
 target_link_libraries(brotlienc brotlicommon)
-endif()
-
-target_link_libraries(brotlidec-static brotlicommon-static)
-target_link_libraries(brotlienc-static brotlicommon-static)
 
 # For projects stuck on older versions of CMake, this will set the
 # BROTLI_INCLUDE_DIRS and BROTLI_LIBRARIES variables so they still
@@ -226,12 +228,12 @@
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   )
 
-  install(
-    TARGETS ${BROTLI_LIBRARIES_CORE}
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-  )
+#  install(
+#    TARGETS ${BROTLI_LIBRARIES_CORE}
+#    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+#    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+#    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+#  )
 
   install(
     TARGETS ${BROTLI_LIBRARIES_CORE_STATIC}
diff -Nru brotli-1.0.9.orig/c/common/platform.h brotli-1.0.9/c/common/platform.h
--- brotli-1.0.9.orig/c/common/platform.h	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/c/common/platform.h	2022-07-10 15:34:19.368776746 +0000
@@ -29,7 +29,7 @@
 #include <brotli/types.h>
 
 #if defined(OS_LINUX) || defined(OS_CYGWIN) || defined(__EMSCRIPTEN__)
-#include <endian.h>
+//#include <endian.h>
 #elif defined(OS_FREEBSD)
 #include <machine/endian.h>
 #elif defined(OS_MACOSX)
diff -Nru brotli-1.0.9.orig/scripts/libbrotlicommon.pc.in brotli-1.0.9/scripts/libbrotlicommon.pc.in
--- brotli-1.0.9.orig/scripts/libbrotlicommon.pc.in	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/scripts/libbrotlicommon.pc.in	2022-07-10 17:05:51.845405730 +0000
@@ -7,5 +7,5 @@
 URL: https://github.com/google/brotli
 Description: Brotli common dictionary library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlicommon
+Libs: -L${libdir} -lbrotlicommon
 Cflags: -I${includedir}
diff -Nru brotli-1.0.9.orig/scripts/libbrotlidec.pc.in brotli-1.0.9/scripts/libbrotlidec.pc.in
--- brotli-1.0.9.orig/scripts/libbrotlidec.pc.in	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/scripts/libbrotlidec.pc.in	2022-07-10 17:06:06.013337085 +0000
@@ -7,6 +7,6 @@
 URL: https://github.com/google/brotli
 Description: Brotli decoder library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlidec
+Libs: -L${libdir} -lbrotlidec
 Requires.private: libbrotlicommon >= 1.0.2
 Cflags: -I${includedir}
diff -Nru brotli-1.0.9.orig/scripts/libbrotlienc.pc.in brotli-1.0.9/scripts/libbrotlienc.pc.in
--- brotli-1.0.9.orig/scripts/libbrotlienc.pc.in	2020-08-27 14:12:55.000000000 +0000
+++ brotli-1.0.9/scripts/libbrotlienc.pc.in	2022-07-10 17:06:19.025273824 +0000
@@ -7,6 +7,6 @@
 URL: https://github.com/google/brotli
 Description: Brotli encoder library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlienc
+Libs: -L${libdir} -lbrotlienc
 Requires.private: libbrotlicommon >= 1.0.2
 Cflags: -I${includedir}
