--- libcss-0.8.0/include/libcss/computed.h.orig	2018-08-22 15:27:44.000000000 +0200
+++ libcss-0.8.0/include/libcss/computed.h	2018-12-29 19:02:55.633399014 +0100
@@ -79,13 +79,13 @@
 css_error css_computed_style_destroy(css_computed_style *style);
 
 css_error css_computed_style_compose(
-		const css_computed_style *restrict parent,
-		const css_computed_style *restrict child,
+		const css_computed_style *__restrict parent,
+		const css_computed_style *__restrict child,
 		css_error (*compute_font_size)(void *pw,
 				const struct css_hint *parent,
 				struct css_hint *size),
 		void *pw,
-		css_computed_style **restrict result);
+		css_computed_style **__restrict result);
 
 /******************************************************************************
  * Property accessors below here                                              *
